c10/core/TensorImpl.h

```diff
@@ -561,6 +561,10 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return strides_default();
   }
 
+  c10::SymIntArrayRef  sym_sizes() const {
+    return c10::SymIntArrayRef(reinterpret_cast<const c10::SymInt*>(sizes_and_strides_.sizes_data()), sizes_and_strides_.size());
```

Comment:
@suo after your patch `sizes_and_strides_` would directly contain SymInt, so this reinterpret cast would be removed